Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -96,24 +96,28 @@ protected override void ProcessRecord()
             switch (ParameterSetName)
             {
                 case ParameterSetNames.List:
-                    DatasetsResource.ListRequest lRequest = new DatasetsResource.ListRequest(Service, Project);
+                    DatasetsResource.ListRequest lRequest = Service.Datasets.List(Project);
```

Comment:
This is getting to be a pretty long case statement. Consider moving most of this code into a function. The standard practice has been to have the function that returns an Enumeration of the objects we want to write to the pipeline, and have the WriteObject call remain in the case statement. Look at how yield return works to help build the Enumeration.

    case ParameterSetNames.List:
        Enumeration<Dataset> datasets = ListProjectDatasets(Project);
        WriteObject(datasets, true);
        break;
    case ParameterSetNames.Get:
        Dataset dataset = GetDataset(Project, Dataset);
        WriteObject(dataset);
        break;